File: app/src/components/DashboardUI.js
// Dashboard tasarım kiti — futbol analiz paneli hissi. Koyu "hero" panel + açık
// metrik kartları + bölümler + görünüm modu + filtre çubuğu + boş durum + barlar.
// Tüm renkler temadan; hem Kullanıcı Paneli hem Sistem Karnesi bunu kullanır.
import React from 'react';
import { View, Text, ScrollView, TouchableOpacity, StyleSheet } from 'react-native';
import { colors, spacing, radius, shadow } from '../theme';

const toneColor = { primary: colors.primary, success: colors.success, warning: colors.warning, danger: colors.danger, info: colors.info, neutral: colors.textSoft };

// ——— Koyu hero panel: başlık + alt metin + öne çıkan metrikler ———
export function DashboardHero({ title, subtitle, kicker, metrics = [], right }) {
  return (
    <View style={s.hero}>
      <View style={s.heroTopRow}>
        <View style={{ flex: 1 }}>
          {kicker ? <Text style={s.heroKicker}>{kicker}</Text> : null}
          <Text style={s.heroTitle}>{title}</Text>
          {subtitle ? <Text style={s.heroSub}>{subtitle}</Text> : null}
        </View>
        {right}
      </View>
      {metrics.length > 0 && (
        <View style={s.heroMetrics}>
          {metrics.map((m, i) => (
            <View key={i} style={[s.heroMetric, i > 0 && s.heroMetricBorder]}>
              <Text style={[s.heroVal, m.tone && { color: toneColor[m.tone] || m.tone }]}>{m.value}</Text>
              <Text style={s.heroLabel} numberOfLines={1}>{m.label}</Text>
            </View>
          ))}
        </View>
      )}
    </View>
  );
}

// ——— Açık metrik kartı: değer + etiket + küçük açıklama ———
export function DashboardMetric({ value, label, hint, tone = 'primary', icon, style }) {
  const c = toneColor[tone] || tone;
  return (
    <View style={[s.metric, style]}>
      <View style={s.metricTop}>
        {icon ? <View style={[s.metricIcon, { backgroundColor: c + '18' }]}><Text style={[s.metricIconTxt, { color: c }]}>{icon}</Text></View> : null}
        <Text style={[s.metricVal, { color: c }]} numberOfLines={1}>{value}</Text>
      </View>
      <Text style={s.metricLabel} numberOfLines={1}>{label}</Text>
      {hint ? <Text style={s.metricHint} numberOfLines={2}>{hint}</Text> : null}
    </View>
  );
}

// ——— Bölüm başlığı ———
export function DashboardSection({ title, count, sub, right, danger }) {
  return (
    <View style={s.section}>
      <View style={s.sectionRow}>
        <Text style={s.sectionTitle}>{title}{count != null ? <Text style={[s.sectionCount, danger && { color: colors.danger }]}>  {count}</Text> : null}</Text>
        {right}
      </View>
      {sub ? <Text style={s.sectionSub}>{sub}</Text> : null}
    </View>
  );
}

// ——— Görünüm modu: Sade / Detaylı / Teknik ———
const VIEW_MODES = [{ k: 'simple', l: 'Sade' }, { k: 'detailed', l: 'Detaylı' }, { k: 'technical', l: 'Teknik' }];
export function ViewModeToggle({ value, onChange }) {
  return (
    <View style={s.segWrap}>
      {VIEW_MODES.map((m) => {
        const on = value === m.k;
        return (
          <TouchableOpacity key={m.k} onPress={() => onChange(m.k)} style={[s.seg, on && s.segOn]} activeOpacity={0.8}>
            <Text style={[s.segTxt, on && s.segTxtOn]}>{m.l}</Text>
          </TouchableOpacity>
        );
      })}
    </View>
  );
}

// ——— Yatay filtre çubuğu ———
export function FilterBar({ options = [], value, onChange }) {
  return (
    <ScrollView horizontal showsHorizontalScrollIndicator={false} contentContainerStyle={s.filterRow}>
      {options.map((o) => {
        const on = value === o.key;
        return (
          <TouchableOpacity key={o.key} onPress={() => onChange(o.key)} style={[s.chip, on && s.chipOn]} activeOpacity={0.8}>
            <Text style={[s.chipTxt, on && s.chipTxtOn]}>{o.label}</Text>
          </TouchableOpacity>
        );
      })}
    </ScrollView>
  );
}

// ——— Etiketli oran barı ———
export function MetricBar({ label, value, total, color, suffix = '%' }) {
  const c = color || (value >= 60 ? colors.success : value >= 45 ? colors.warning : colors.danger);
  return (
    <View style={s.bar}>
      <View style={s.barTop}>
        <Text style={s.barLabel} numberOfLines={1}>{label}{total != null ? <Text style={s.barTotal}>  ({total})</Text> : null}</Text>
        <Text style={[s.barPct, { color: c }]}>{value}{suffix}</Text>
      </View>
      <View style={s.barTrack}><View style={[s.barFill, { width: `${Math.max(0, Math.min(100, value))}%`, backgroundColor: c }]} /></View>
    </View>
  );
}

// ——— Profesyonel boş durum ———
export function DashboardEmpty({ icon = '📊', title, message, actionLabel, onAction }) {
  return (
    <View style={s.empty}>
      <Text style={s.emptyIcon}>{icon}</Text>
      <Text style={s.emptyTitle}>{title}</Text>
      {message ? <Text style={s.emptyMsg}>{message}</Text> : null}
      {actionLabel && onAction ? (
        <TouchableOpacity onPress={onAction} style={s.emptyBtn} activeOpacity={0.85}><Text style={s.emptyBtnTxt}>{actionLabel}</Text></TouchableOpacity>
      ) : null}
    </View>
  );
}

const s = StyleSheet.create({
  hero: { backgroundColor: colors.primary, borderRadius: radius.xl, padding: spacing.lg, ...shadow.card },
  heroTopRow: { flexDirection: 'row', alignItems: 'flex-start' },
  heroKicker: { color: '#8FA6C9', fontSize: 10.5, fontWeight: '900', letterSpacing: 1.2, textTransform: 'uppercase' },
  heroTitle: { color: '#FFFFFF', fontSize: 21, fontWeight: '900', marginTop: 3 },
  heroSub: { color: '#B7C4DA', fontSize: 12, fontWeight: '600', marginTop: 4, lineHeight: 16 },
  heroMetrics: { flexDirection: 'row', marginTop: spacing.lg, backgroundColor: colors.darkCardSoft, borderRadius: radius.md, paddingVertical: 12 },
  heroMetric: { flex: 1, alignItems: 'center', paddingHorizontal: 6 },
  heroMetricBorder: { borderLeftWidth: 1, borderLeftColor: '#2A3A57' },
  heroVal: { color: '#FFFFFF', fontSize: 22, fontWeight: '900' },
  heroLabel: { color: '#9DB0CD', fontSize: 10, fontWeight: '700', marginTop: 3, textAlign: 'center' },

  metric: { flex: 1, backgroundColor: colors.card, borderRadius: radius.lg, borderWidth: 1, borderColor: colors.border, padding: 13, ...shadow.soft },
  metricTop: { flexDirection: 'row', alignItems: 'center', gap: 8 },
  metricIcon: { width: 26, height: 26, borderRadius: 8, alignItems: 'center', justifyContent: 'center' },
  metricIconTxt: { fontSize: 13, fontWeight: '900' },
  metricVal: { fontSize: 22, fontWeight: '900', flexShrink: 1 },
  metricLabel: { color: colors.textSoft, fontSize: 12, fontWeight: '800', marginTop: 8 },
  metricHint: { color: colors.textMuted, fontSize: 10.5, fontWeight: '600', marginTop: 3, lineHeight: 14 },

  section: { marginTop: spacing.xl, marginBottom: spacing.sm },
  sectionRow: { flexDirection: 'row', alignItems: 'center', justifyContent: 'space-between' },
  sectionTitle: { color: colors.text, fontSize: 16, fontWeight: '900' },
  sectionCount: { color: colors.textMuted, fontSize: 15, fontWeight: '900' },
  sectionSub: { color: colors.textMuted, fontSize: 11.5, fontWeight: '600', marginTop: 3 },

  segWrap: { flexDirection: 'row', backgroundColor: colors.cardAlt, borderRadius: radius.md, padding: 3, gap: 2 },
  seg: { paddingHorizontal: 12, paddingVertical: 6, borderRadius: radius.sm },
  segOn: { backgroundColor: colors.card, ...shadow.soft },
  segTxt: { color: colors.textMuted, fontSize: 12, fontWeight: '800' },
  segTxtOn: { color: colors.primary },

  filterRow: { gap: 7, paddingVertical: 2 },
  chip: { paddingHorizontal: 13, paddingVertical: 7, borderRadius: radius.pill, backgroundColor: colors.cardAlt, borderWidth: 1, borderColor: colors.border },
  chipOn: { backgroundColor: colors.primary, borderColor: colors.primary },
  chipTxt: { color: colors.textSoft, fontSize: 12, fontWeight: '800' },
  chipTxtOn: { color: '#FFFFFF' },

  bar: { marginBottom: 11 },
  barTop: { flexDirection: 'row', justifyContent: 'space-between', alignItems: 'center', marginBottom: 5 },
  barLabel: { flex: 1, color: colors.textSoft, fontSize: 12, fontWeight: '700' },
  barTotal: { color: colors.textMuted, fontSize: 11, fontWeight: '700' },
  barPct: { fontSize: 13, fontWeight: '900', marginLeft: 8 },
  barTrack: { height: 8, borderRadius: 5, backgroundColor: colors.bgAlt, overflow: 'hidden' },
  barFill: { height: 8, borderRadius: 5 },

  empty: { flex: 1, alignItems: 'center', justifyContent: 'center', padding: 32, minHeight: 320 },
  emptyIcon: { fontSize: 46 },
  emptyTitle: { color: colors.text, fontSize: 17, fontWeight: '900', marginTop: 12, textAlign: 'center' },
  emptyMsg: { color: colors.textMuted, fontSize: 13, fontWeight: '600', marginTop: 8, textAlign: 'center', lineHeight: 19, maxWidth: 320 },
  emptyBtn: { marginTop: 18, backgroundColor: colors.primary, paddingHorizontal: 22, paddingVertical: 12, borderRadius: radius.md },
  emptyBtnTxt: { color: '#FFFFFF', fontSize: 13.5, fontWeight: '800' },
});

File: flutter/lib/widgets/dashboard_ui.dart
// KAYNAK: app/src/components/DashboardUI.js — BİREBİR çeviri.
//
// Dashboard tasarım kiti — futbol analiz paneli hissi. Koyu "hero" panel + açık
// metrik kartları + bölümler + görünüm modu + filtre çubuğu + boş durum + barlar.
// Tüm renkler temadan; hem Kullanıcı Paneli hem Sistem Karnesi bunu kullanır.

import 'package:flutter/material.dart';

import '../core/theme/tokens.dart';

Color _toneColor(Object? tone) {
  if (tone is Color) return tone;
  return switch (tone) {
    'primary' => AppColors.primary,
    'success' => AppColors.success,
    'warning' => AppColors.warning,
    'danger' => AppColors.danger,
    'info' => AppColors.info,
    'neutral' => AppColors.textSoft,
    _ => AppColors.primary,
  };
}

typedef HeroMetric = ({String value, String label, Object? tone});

/// Koyu hero panel: başlık + alt metin + öne çıkan metrikler.
class DashboardHero extends StatelessWidget {
  // `const` DEĞİL — BİLEREK (2026-08-12): bu widget rengini `AppColors`
  // küresellerinden okuyor ve tema çalışma zamanında değişiyor. `const`
  // yapıcı widget örneğini sabitler; Flutter aynı örneği görünce alt ağacı
  // YENİDEN KURMAZ ve widget eski renkte donar (emülatörde ölçüldü:
  // Dortmund temasında kupon boş-durum kartı Galatasaray bordosunda kaldı).
  // ignore: prefer_const_constructors_in_immutables
  DashboardHero({
    super.key,
    required this.title,
    this.subtitle,
    this.kicker,
    this.metrics = const [],
    this.right,
  });

  final String title;
  final String? subtitle;
  final String? kicker;
  final List<HeroMetric> metrics;
  final Widget? right;

  @override
  Widget build(BuildContext context) => Container(
    padding: const EdgeInsets.all(Spacing.lg),
    decoration: BoxDecoration(
      // HERO KENDİ TOKENİYLE BOYANIR (16 Ağustos 2026 denetimi).
      //
      // Burada `primary` kullanılıyordu; oysa tema hero için AYRI bir yüzey
      // tanımlıyor (`heroZemin`) ve hero yazıları (`onHero`/`onHeroSoft`) tam
      // O YÜZEYE göre türetiliyor (bkz. takim_gorunumu.dart). Yüzey `primary`
      // olunca yazılar başka bir yüzeyin renginden geliyordu ve takım
      // temasında Sistem Karnesi başlığı okunmuyordu (ölçüldü:
      // `t18_sistem_karnesi.png` — "Sistem Master Analiz Karnesi" silik).
      //
      // Ana sayfadaki hero zaten `heroZemin` kullanıyor; iki hero ayrışmasın.
      color: AppColors.heroZemin,
      borderRadius: BorderRadius.circular(AppRadius.xl),
      boxShadow: AppShadow.card,
    ),
    child: Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Row(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Expanded(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  if (kicker != null)
                    Text(
                      kicker!.toUpperCase(),
                      // Kartın zemini `AppColors.primary` — yazılar ona göre
                      // hesaplanır. Eskiden marka laciverdine göre seçilmiş
                      // sabit grilerdi ve sarı temalı takımda okunmuyordu.
                      style: TextStyle(
                        color: AppColors.onHeroSoft,
                        fontSize: 10.5,
                        fontWeight: AppFont.black,
                        letterSpacing: 1.2,
                      ),
                    ),
                  Padding(
                    padding: const EdgeInsets.only(top: 3),
                    child: Text(
                      title,
                      style: TextStyle(
                        color: AppColors.onHero,
                        fontSize: 21,
                        fontWeight: AppFont.black,
                      ),
                    ),
                  ),
                  if (subtitle != null)
                    Padding(
                      padding: const EdgeInsets.only(top: 4),
                      child: Text(
                        subtitle!,
                        style: TextStyle(
                          color: AppColors.onHeroSoft,
                          fontSize: 12,
                          fontWeight: AppFont.semibold,
                          height: 16 / 12,
                        ),
                      ),
                    ),
                ],
              ),
            ),
            ?right,
          ],
        ),
        if (metrics.isNotEmpty)
          Container(
            margin: const EdgeInsets.only(top: Spacing.lg),
            padding: const EdgeInsets.symmetric(vertical: 12),
            decoration: BoxDecoration(
              color: AppColors.darkCardSoft,
              borderRadius: BorderRadius.circular(AppRadius.md),
            ),
            child: IntrinsicHeight(
              child: Row(
                children: [
                  for (var i = 0; i < metrics.length; i++)
                    Expanded(
                      child: Container(
                        padding: const EdgeInsets.symmetric(horizontal: 6),
                        decoration: i > 0
                            ? BoxDecoration(
                                border: Border(
                                  left: BorderSide(color: AppColors.darkBorder),
                                ),
                              )
                            : null,
                        child: Column(
                          children: [
                            Text(
                              metrics[i].value,
                              style: TextStyle(
                                color: metrics[i].tone != null
                                    ? _toneColor(metrics[i].tone)
                                    : AppColors.onDark,
                                fontSize: 22,
                                fontWeight: AppFont.black,
                              ),
                            ),
                            Padding(
                              padding: const EdgeInsets.only(top: 3),
                              child: Text(
                                metrics[i].label,
                                maxLines: 1,
                                overflow: TextOverflow.ellipsis,
                                textAlign: TextAlign.center,
                                // İç kutunun zemini `darkCardSoft`.
                                style: TextStyle(
                                  color: AppColors.onDarkSoft,
                                  fontSize: 10,
                                  fontWeight: AppFont.bold,
                                ),
                              ),
                            ),
                          ],
                        ),
                      ),
                    ),
                ],
              ),
            ),
          ),
      ],
    ),
  );
}

/// Açık metrik kartı: değer + etiket + küçük açıklama.
class DashboardMetric extends StatelessWidget {
  // `const` DEĞİL — BİLEREK (2026-08-12): bu widget rengini `AppColors`
  // küresellerinden okuyor ve tema çalışma zamanında değişiyor. `const`
  // yapıcı widget örneğini sabitler; Flutter aynı örneği görünce alt ağacı
  // YENİDEN KURMAZ ve widget eski renkte donar (emülatörde ölçüldü:
  // Dortmund temasında kupon boş-durum kartı Galatasaray bordosunda kaldı).
  // ignore: prefer_const_constructors_in_immutables
  DashboardMetric({
    super.key,
    required this.value,
    required this.label,
    this.hint,
    this.tone = 'primary',
    this.icon,
  });

  final String value;
  final String label;
  final String? hint;
  final Object tone;
  final String? icon;

  @override
  Widget build(BuildContext context) {
    final c = _toneColor(tone);
    return Container(
      padding: const EdgeInsets.all(13),
      decoration: BoxDecoration(
        color: AppColors.card,
        borderRadius: BorderRadius.circular(AppRadius.lg),
        border: Border.all(color: AppColors.border),
        boxShadow: AppShadow.soft,
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Row(
            children: [
              if (icon != null) ...[
                Container(
                  width: 26,
                  height: 26,
                  alignment: Alignment.center,
                  decoration: BoxDecoration(
                    // Kaynakta `c + '18'` → rengin %9'u.
                    color: c.withValues(alpha: 0x18 / 0xFF),
                    borderRadius: BorderRadius.circular(8),
                  ),
                  child: Text(
                    icon!,
                    style: TextStyle(
                      color: c,
                      fontSize: 13,
                      fontWeight: AppFont.black,
                    ),
                  ),
                ),
                const SizedBox(width: 8),
              ],
              Flexible(
                child: Text(
                  value,
                  maxLines: 1,
                  overflow: TextOverflow.ellipsis,
                  style: TextStyle(
                    color: c,
                    fontSize: 22,
                    fontWeight: AppFont.black,
                  ),
                ),
              ),
            ],
          ),
          Padding(
            padding: const EdgeInsets.only(top: 8),
            child: Text(
              label,
              maxLines: 1,
              overflow: TextOverflow.ellipsis,
              style: TextStyle(
                color: AppColors.textSoft,
                fontSize: 12,
                fontWeight: AppFont.heavy,
              ),
            ),
          ),
          if (hint != null)
            Padding(
              padding: const EdgeInsets.only(top: 3),
              child: Text(
                hint!,
                maxLines: 2,
                overflow: TextOverflow.ellipsis,
                style: TextStyle(
                  color: AppColors.textMuted,
                  fontSize: 10.5,
                  fontWeight: AppFont.semibold,
                  height: 14 / 10.5,
                ),
              ),
            ),
        ],
      ),
    );
  }
}

/// Bölüm başlığı.
class DashboardSection extends StatelessWidget {
  // `const` DEĞİL — BİLEREK (2026-08-12): bu widget rengini `AppColors`
  // küresellerinden okuyor ve tema çalışma zamanında değişiyor. `const`
  // yapıcı widget örneğini sabitler; Flutter aynı örneği görünce alt ağacı
  // YENİDEN KURMAZ ve widget eski renkte donar (emülatörde ölçüldü:
  // Dortmund temasında kupon boş-durum kartı Galatasaray bordosunda kaldı).
  // ignore: prefer_const_constructors_in_immutables
  DashboardSection({
    super.key,
    required this.title,
    this.count,
    this.sub,
    this.right,
    this.danger = false,
  });

  final String title;
  final Object? count;
  final String? sub;
  final Widget? right;
  final bool danger;

  @override
  Widget build(BuildContext context) => Padding(
    padding: const EdgeInsets.only(top: Spacing.xl, bottom: Spacing.sm),
    child: Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Row(
          mainAxisAlignment: MainAxisAlignment.spaceBetween,
          children: [
            Flexible(
              child: RichText(
                text: TextSpan(
                  style: TextStyle(
                    color: AppColors.text,
                    fontSize: 16,
                    fontWeight: AppFont.black,
                  ),
                  children: [
                    TextSpan(text: title),
                    if (count != null)
                      TextSpan(
                        text: '  $count',
                        style: TextStyle(
                          color: danger
                              ? AppColors.danger
                              : AppColors.textMuted,
                          fontSize: 15,
                          fontWeight: AppFont.black,
                        ),
                      ),
                  ],
                ),
              ),
            ),
            ?right,
          ],
        ),
        if (sub != null)
          Padding(
            padding: const EdgeInsets.only(top: 3),
            child: Text(
              sub!,
              style: TextStyle(
                color: AppColors.textMuted,
                fontSize: 11.5,
                fontWeight: AppFont.semibold,
              ),
            ),
          ),
      ],
    ),
  );
}

/// Görünüm modu: Sade / Detaylı / Teknik.
class ViewModeToggle extends StatelessWidget {
  // `const` DEĞİL — BİLEREK (2026-08-12): bu widget rengini `AppColors`
  // küresellerinden okuyor ve tema çalışma zamanında değişiyor. `const`
  // yapıcı widget örneğini sabitler; Flutter aynı örneği görünce alt ağacı
  // YENİDEN KURMAZ ve widget eski renkte donar (emülatörde ölçüldü:
  // Dortmund temasında kupon boş-durum kartı Galatasaray bordosunda kaldı).
  // ignore: prefer_const_constructors_in_immutables
  ViewModeToggle({super.key, required this.value, required this.onChange});

  final String? value;
  final ValueChanged<String> onChange;

  static const List<({String k, String l})> _modes = [
    (k: 'simple', l: 'Sade'),
    (k: 'detailed', l: 'Detaylı'),
    (k: 'technical', l: 'Teknik'),
  ];

  @override
  Widget build(BuildContext context) => Container(
    padding: const EdgeInsets.all(3),
    decoration: BoxDecoration(
      color: AppColors.cardAlt,
      borderRadius: BorderRadius.circular(AppRadius.md),
    ),
    child: Row(
      mainAxisSize: MainAxisSize.min,
      children: [
        for (var i = 0; i < _modes.length; i++) ...[
          if (i > 0) const SizedBox(width: 2),
          GestureDetector(
            onTap: () => onChange(_modes[i].k),
            behavior: HitTestBehavior.opaque,
            child: Container(
              padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 6),
              decoration: value == _modes[i].k
                  ? BoxDecoration(
                      color: AppColors.card,
                      borderRadius: BorderRadius.circular(AppRadius.sm),
                      boxShadow: AppShadow.soft,
                    )
                  : null,
              child: Text(
                _modes[i].l,
                style: TextStyle(
                  color: value == _modes[i].k
                      ? AppColors.primary
                      : AppColors.textMuted,
                  fontSize: 12,
                  fontWeight: AppFont.heavy,
                ),
              ),
            ),
          ),
        ],
      ],
    ),
  );
}

/// Yatay filtre çubuğu.
class FilterBar extends StatelessWidget {
  // `const` DEĞİL — BİLEREK (2026-08-12): bu widget rengini `AppColors`
  // küresellerinden okuyor ve tema çalışma zamanında değişiyor. `const`
  // yapıcı widget örneğini sabitler; Flutter aynı örneği görünce alt ağacı
  // YENİDEN KURMAZ ve widget eski renkte donar (emülatörde ölçüldü:
  // Dortmund temasında kupon boş-durum kartı Galatasaray bordosunda kaldı).
  // ignore: prefer_const_constructors_in_immutables
  FilterBar({
    super.key,
    required this.options,
    required this.value,
    required this.onChange,
  });

  final List<({String key, String label})> options;
  final String? value;
  final ValueChanged<String> onChange;

  @override
  Widget build(BuildContext context) => SingleChildScrollView(
    scrollDirection: Axis.horizontal,
    padding: const EdgeInsets.symmetric(vertical: 2),
    child: Row(
      children: [
        for (var i = 0; i < options.length; i++) ...[
          if (i > 0) const SizedBox(width: 7),
          GestureDetector(
            onTap: () => onChange(options[i].key),
            behavior: HitTestBehavior.opaque,
            child: Container(
              padding: const EdgeInsets.symmetric(horizontal: 13, vertical: 7),
              decoration: BoxDecoration(
                // SEÇİLİ ÇİP ZEMİNDE DURUYOR (16 Ağustos 2026 denetimi).
                //
                // Dolgu `primary`ydi; takım temasının "birinci renk" modunda
                // sayfa zemini de o aileden olduğu için seçili çip kayboluyor,
                // sekme adı yalın yazı gibi kalıyordu (ölçüldü:
                // `t18_sistem_karnesi.png` — "Özet" çipsiz görünüyor).
                // Kullanıcı kararıyla aynı dil: KART dolgusu, SARI (primary)
                // çerçeve ve yazı.
                color: value == options[i].key
                    ? AppColors.card
                    : AppColors.cardAlt,
                borderRadius: BorderRadius.circular(AppRadius.pill),
                border: Border.all(
                  color: value == options[i].key
                      ? AppColors.primary
                      : AppColors.border,
                  width: value == options[i].key ? 1.5 : 1,
                ),
              ),
              child: Text(
                options[i].label,
                style: TextStyle(
                  color: value == options[i].key
                      ? AppColors.primary
                      : AppColors.textSoft,
                  fontSize: 12,
                  fontWeight: AppFont.heavy,
                ),
              ),
            ),
          ),
        ],
      ],
    ),
  );
}

/// Etiketli oran barı.
///
/// Renk eşiği kaynaktan: %60+ yeşil, %45+ sarı, altı kırmızı. Bu renk bir
/// "kazanır/kaybeder" iddiası DEĞİL, yalnız ölçünün nerede durduğudur.
class MetricBar extends StatelessWidget {
  // `const` DEĞİL — BİLEREK (2026-08-12): bu widget rengini `AppColors`
  // küresellerinden okuyor ve tema çalışma zamanında değişiyor. `const`
  // yapıcı widget örneğini sabitler; Flutter aynı örneği görünce alt ağacı
  // YENİDEN KURMAZ ve widget eski renkte donar (emülatörde ölçüldü:
  // Dortmund temasında kupon boş-durum kartı Galatasaray bordosunda kaldı).
  // ignore: prefer_const_constructors_in_immutables
  MetricBar({
    super.key,
    required this.label,
    required this.value,
    this.total,
    this.color,
    this.suffix = '%',
  });

  final String label;
  final num value;
  final Object? total;
  final Color? color;
  final String suffix;

  @override
  Widget build(BuildContext context) {
    final c =
        color ??
        (value >= 60
            ? AppColors.success
            : (value >= 45 ? AppColors.warning : AppColors.danger));
    final safe = value.clamp(0, 100).toDouble();
    return Padding(
      padding: const EdgeInsets.only(bottom: 11),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Row(
            children: [
              Expanded(
                child: RichText(
                  maxLines: 1,
                  overflow: TextOverflow.ellipsis,
                  text: TextSpan(
                    style: TextStyle(
                      color: AppColors.textSoft,
                      fontSize: 12,
                      fontWeight: AppFont.bold,
                    ),
                    children: [
                      TextSpan(text: label),
                      if (total != null)
                        TextSpan(
                          text: '  ($total)',
                          style: TextStyle(
                            color: AppColors.textMuted,
                            fontSize: 11,
                            fontWeight: AppFont.bold,
                          ),
                        ),
                    ],
                  ),
                ),
              ),
              const SizedBox(width: 8),
              Text(
                '$value$suffix',
                style: TextStyle(
                  color: c,
                  fontSize: 13,
                  fontWeight: AppFont.black,
                ),
              ),
            ],
          ),
          const SizedBox(height: 5),
          ClipRRect(
            borderRadius: BorderRadius.circular(5),
            child: Container(
              height: 8,
              color: AppColors.bgAlt,
              child: FractionallySizedBox(
                alignment: Alignment.centerLeft,
                widthFactor: safe / 100,
                child: ColoredBox(color: c),
              ),
            ),
          ),
        ],
      ),
    );
  }
}

/// Profesyonel boş durum.
class DashboardEmpty extends StatelessWidget {
  // `const` DEĞİL — BİLEREK (2026-08-12): bu widget rengini `AppColors`
  // küresellerinden okuyor ve tema çalışma zamanında değişiyor. `const`
  // yapıcı widget örneğini sabitler; Flutter aynı örneği görünce alt ağacı
  // YENİDEN KURMAZ ve widget eski renkte donar (emülatörde ölçüldü:
  // Dortmund temasında kupon boş-durum kartı Galatasaray bordosunda kaldı).
  // ignore: prefer_const_constructors_in_immutables
  DashboardEmpty({
    super.key,
    this.icon = Icons.bar_chart,
    required this.title,
    this.message,
    this.actionLabel,
    this.onAction,
  });

  /// VEKTÖR ikon, emoji DEĞİL (kullanıcı isteği, 2026-08-12).
  final IconData icon;
  final String title;
  final String? message;
  final String? actionLabel;
  final VoidCallback? onAction;

  @override
  Widget build(BuildContext context) => Container(
    constraints: const BoxConstraints(minHeight: 320),
    padding: const EdgeInsets.all(32),
    alignment: Alignment.center,
    child: Column(
      mainAxisSize: MainAxisSize.min,
      children: [
        Icon(icon, size: 48, color: AppColors.textSoft),
        Padding(
          padding: const EdgeInsets.only(top: 12),
          child: Text(
            title,
            textAlign: TextAlign.center,
            style: TextStyle(
              color: AppColors.text,
              fontSize: 17,
              fontWeight: AppFont.black,
            ),
          ),
        ),
        if (message != null)
          Padding(
            padding: const EdgeInsets.only(top: 8),
            child: ConstrainedBox(
              constraints: const BoxConstraints(maxWidth: 320),
              child: Text(
                message!,
                textAlign: TextAlign.center,
                style: TextStyle(
                  color: AppColors.textMuted,
                  fontSize: 13,
                  fontWeight: AppFont.semibold,
                  height: 19 / 13,
                ),
              ),
            ),
          ),
        if (actionLabel != null && onAction != null)
          Padding(
            padding: const EdgeInsets.only(top: 18),
            child: GestureDetector(
              onTap: onAction,
              child: Container(
                padding: const EdgeInsets.symmetric(
                  horizontal: 22,
                  vertical: 12,
                ),
                decoration: BoxDecoration(
                  color: AppColors.primary,
                  borderRadius: BorderRadius.circular(AppRadius.md),
                ),
                child: Text(
                  actionLabel!,
                  style: TextStyle(
                    color: AppColors.onPrimary,
                    fontSize: 13.5,
                    fontWeight: AppFont.heavy,
                  ),
                ),
              ),
            ),
          ),
      ],
    ),
  );
}
